api/classes/class-lightweight-api.php

```diff
@@ -130,16 +130,17 @@ public function error( $code ) {
 	 */
 	public function get_transient( $name ) {
 		global $wpdb;
-		$name = '_transient_' . $name;
-
-		$value = wp_cache_get( $name, 'newspack-popups' );
+		$name       = '_transient_' . $name;
+		$table_name = Segmentation::get_transients_table_name();
+		$value      = wp_cache_get( $name, 'newspack-popups' );
 		if ( -1 === $value ) {
 			$this->debug['read_empty_transients'] += 1;
 			$this->debug['cache_count']           += 1;
 			return null;
 		} elseif ( false === $value ) {
 			$this->debug['read_query_count'] += 1;
-			$value                            = $this->get_option( $name );
+
+			$value = $wpdb->get_var( $wpdb->prepare( 'SELECT option_value FROM %s WHERE option_name = %s LIMIT 1', $table_name, $name ) ); // phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching;
```

Comment:
Not sure if that's maybe related to my local setup, but I was getting a ` WordPress database error You have an error in your SQL syntax` error because of bad escaping of table name. Here's what fixed it:

```suggestion
			$value = $wpdb->get_var( $wpdb->prepare( "SELECT option_value FROM `$table_name` WHERE option_name = %s LIMIT 1", $name ) ); // phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching;
```

Didn't you encounter the same issue?

Reply:
Strange, I didn't encounter this error when testing this PR.

Reply:
FWIW I'm running PHP `7.4.1` & MySQL `8.0.16`

Reply:
Me neither, but the fix looks good: https://github.com/Automattic/newspack-popups/pull/488/commits/2b7ede234bdf9068aec71e4c42230122c911a0a5

Reply:
Also running PHP 7.4.1 and MySQL 8.0.16

Reply:
> Me neither, but the fix looks good: 2b7ede2

Almost! PHP needs double quotes to interpolate a string.

Reply:
Fixed in 5f52d6a0225c106bdc12b0b60563afdb09138902